runtime/src/plugin.rs

```diff
@@ -290,24 +306,34 @@ impl Plugin {
                 (entry.0.as_str(), entry.1)
             });
 
-        // Avoid running into resource limits, after 5 instantiations reset the store. This will
-        // release any old `Instance` objects
         if self.instantiations > 5 {
-            self.memory.get_mut().reinstantiate()?;
+            let engine = self.store.engine().clone();
+            let internal = self.internal();
+            self.store = Store::new(
+                &engine,
+                Internal::new(
+                    internal.manifest.clone(),
+                    internal.wasi.is_some(),
+                    internal.available_pages,
+                )?,
+            );
+            self.store
+                .epoch_deadline_callback(|_internal| Err(Error::msg("timeout")));
```

Comment:
related to the above - just curious if this is something we were going to remove & have implemented elsewhere.